examples/test-exec-doc.yml

```diff
@@ -0,0 +1,221 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
+  workflow_dispatch:        # Allows manual triggering of the workflow
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest  # Run the job on the latest Ubuntu runner
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0        # Fetch all history for accurate diff
+
+    - name: Check for changed Markdown files
+      id: check_changes
+      run: |
+        # Check if this is the initial commit (all zeros SHA)
+        if [ "${{ github.event.before }}" = "0000000000000000000000000000000000000000" ]; then
```

Comment:
Isn't the only scenario where this action could run on an initial commit is if the repository is initialized with both markdown files and this action? How likely is that and is this check needed?

Reply:
it would run on an initial commit if a new branch is created and the first commit is made. in that case, it checks for changed files by comparing it to the latest commit of its upstream branch so if no changes are made from that last commit, it skips over this step. if there are changes, it tracks only those docs

Reply:
Not quite sure that I follow, could you illustrate an example of how someone would encounter this?

Reply:
as a user, say I created a new branch off of main and didn't make any changes to any markdown files and created a commit, then this will compare the diff with the last commit of main and see there are no unchanged markdown files and not test anything

if I create a new branch off of main and change file1.md, file2.md then it will compare this commit with the last commit of main and see 2 markdown files changed

if I am updating my new branch by changing file1.md, then it will compare the commit to the new branch's last commit and only test file1.md

and if I create a new branch off of something other than main/master (default branches), it tests all the docs there.